pkl-server/src/main/kotlin/org/pkl/server/Server.kt

```diff
@@ -21,13 +21,16 @@ import java.net.URI
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.ExecutorService
 import java.util.concurrent.Executors
+import java.util.regex.Pattern
 import kotlin.random.Random
 import org.pkl.core.*
-import org.pkl.core.evaluatorSettings.PklEvaluatorSettings.ExternalReader
-import org.pkl.core.externalreader.ExternalReaderProcess
+import org.pkl.core.evaluatorSettings.PklEvaluatorSettings
+import org.pkl.core.externalreader.ModuleReaderSpec
+import org.pkl.core.externalreader.ReaderProcess
+import org.pkl.core.externalreader.ResourceReaderSpec
```

Comment:
Given that the resolver types are shared between the client readers and external readers, is org.pkl.core.externalreader really the right place for these types?

Reply:
`org.pkl.server` depending on `externalreader` seems acceptable to me. These types couldn't stay in `module`/`resource` because that creates package cycles and is a layering violation (`module`/`resource` already export `packagereader`, hence `packagereader` shouldn't also depend on them).